### PR TITLE
JAMES-2393 PLUGINS modules were not registered for Cassandra LDAP

### PR DESCRIPTION
**Pull request by chibenwa**
Structurally solve this by exposing the right module aggregate (like for memory)

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please (yoda)

File: mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/CassandraJamesSmtpHostSystem.java [hunk 1]
```diff
@@ -109,7 +110,7 @@ protected GuiceJamesServer createJamesServer() throws Exception {
             .build();
 
         return new GuiceJamesServer(configuration)
-            .combineWith(CassandraJamesServerMain.CASSANDRA_SERVER_MODULE, CassandraJamesServerMain.PROTOCOLS, new ProtocolHandlerModule())
+            .combineWith(CASSANDRA_MODULE_AGGREGATE, new ProtocolHandlerModule())
```
Comment (mbaechler) [inline mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/CassandraJamesSmtpHostSystem.java]: I think SMTP-only don't need quota plugins

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java [hunk 2]
```diff
@@ -98,10 +98,15 @@
         new TikaMailboxModule(),
         new SpamAssassinListenerModule());
 
+    public static Module CASSANDRA_MODULE_AGGREGATE = Modules.combine(
```
Comment (mbaechler) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java]: Module name is not ok, it fails to describe what it aggregates

Comment (chibenwa) [reply server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java]: Could you propose a better name?

File: mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/CassandraJamesSmtpHostSystem.java (hunk 1 above)
Comment (chibenwa) [reply mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/CassandraJamesSmtpHostSystem.java]: Shouldn't we play our integration test on a whole system?

Does it protect us from forgetting plugin registration in the future?

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java (hunk 2 above)
Comment (chibenwa) [reply server/container/guice/cassandra-guice/src/main/java/org/apache/james/CassandraJamesServerMain.java]: BTW we have similar terms in other places: `IN_MEMORY_SERVER_AGGREGATE_MODULE` for memory.

File: mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/CassandraJamesSmtpHostSystem.java (hunk 1 above)
Comment (mbaechler) [reply mpt/impl/smtp/cassandra/src/test/java/org/apache/james/mpt/smtp/host/CassandraJamesSmtpHostSystem.java]: In my understanding, this system is used only for smtp test without mailbox storage.
Quota is about storage so yes, it's an integration test, but not of the IMAP + SMTP combination AFAIK.

**Comment by aduprat** [comment]
```
Tests in error: 
  SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxOrInboxWhenMultipleRecipientsConfigurations:608 » ConditionTimeout
```
test this please

**Comment by aduprat** [comment]
```
Tests in error: 
  SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxOrInboxWhenMultipleRecipientsConfigurations:608 » ConditionTimeout
```
test this please

**Comment by aduprat** [comment]
```
Error executing docker rm --force=true --volumes=true git-publish-james-2393-fix-53d5c5f6-48a9-44fa-9b0c-0b9b43d222ae : no known node for /home/jenkins/build/workspace/571
```
test this please

**Comment by aduprat** [comment]
test this please (yoda)

**Comment by aduprat** [comment]
```
Failed tests: 
  ElasticSearchIndexerTest.deleteByQueryShouldWorkWhenMultipleMessages:182 expected:<[1]L> but was:<[3]L>
```
test this please

**Comment by aduprat** [comment]
```
Failed tests: 
  ElasticSearchIndexerTest.deleteByQueryShouldWorkWhenMultipleMessages:182 expected:<[1]L> but was:<[2]L>
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**